lib/blockchain.go

```diff
@@ -2633,15 +2640,172 @@ func (bc *Blockchain) processBlockPoW(desoBlock *MsgDeSoBlock, verifySignatures
 }
 
 // processBlockPoS runs the Fast-Hotstuff block connect and commit rule as follows:
```

Comment:
What do you think of moving the `ProcessBlockPoS` function to a new `pos_blockchain.go` file? Seems like a tossup since it splits a single struct's implementation into two files, but the `blockchain.go` file is already massive. 

Up to you

Reply:
I'm moving all of the logic to `pos_blockchain.go` (including the validations) so it can all live in one easy place.